2.2: Bump Marathon to 1.11.23 (#7783)
dcos/dcos: CHANGES.md

* Enforce-role property no longer shows for sub-groups (MARATHON-8769)

* Fix issue in which PUT /v2/groups ignored the enforceRole setting (MARATHON-8770)

> * Marathon can send placement constraint signals to Mesos in order to unusable offers (MARATHON-8764)

Reviewer: please rephrase!

> * Marathon can send placement constraint signals to Mesos in order to unusable offers (MARATHON-8764)

Reviewer: ```suggestion
* Marathon can send offer constraints to Mesos to reduce the number of offers Marathon needs to decline due to placement constraints (MARATHON-8764)
```

* Marathon will allow GROUP_BY placement constraints to be temporarily imbalanced during deployments. (MARATHON-8752)

> * Marathon no longer logs task data, which was a source of secrets for Docker containers to be logged. (MARATHON-8774)

Reviewer: ```suggestion
* Marathon no longer logs task data, as it may contain secrets for Docker containers. (MARATHON-8774)
```

Reviewer: just a suggestion
